src/litdata/processing/utilities.py

```diff
@@ -177,3 +189,63 @@ def _get_work_dir() -> str:
     assert project_id is not None
     assert work_id is not None
     return f"s3://{bucket_name}/projects/{project_id}/lightningapps/{app_id}/artifacts/{work_id}/content/"
+
+
+def read_index_file_content(output_dir: Dir) -> Optional[Dict[str, Any]]:
+    """Read the index file content."""
+    if not isinstance(output_dir, Dir):
+        raise ValueError("The provided output_dir should be a Dir object.")
+
+    if output_dir.url is None:
+        if output_dir.path is None:
+            return None
+        index_file_path = os.path.join(output_dir.path, _INDEX_FILENAME)
+        if not os.path.exists(index_file_path):
+            return None
+        with open(index_file_path) as f:
+            return json.load(f)
+
+    else:
+        # download the index file from s3, and read it
+        obj = parse.urlparse(output_dir.url)
+
+        if obj.scheme != "s3":
```

Comment:
Note for another PR. We support GCP in a minimal way, so it would be good to make this more general. As suggested in one of the issues, we could use fsspec.